File: Array/121_BestTimeToBuyAndSellStock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice = prices[0]
        maxprofit = 0
        for price in prices:
            maxprofit = max(maxprofit, price - minprice)
            minprice = min(minprice, price)
        return maxprofit

File: Java Solutions/121_BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int lowest = prices[0];
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < lowest) {
                lowest = prices[i];
            } else if (prices[i] - lowest > res) {
                res = prices[i] - lowest;
            }
        }
        return res;
    }
}